sdk/test-utils/recorder/GUIDELINES.md

```diff
@@ -0,0 +1,333 @@
+
+  For example
+
+  ```typescript
+  describe("Aborter", () => {
+    beforeEach(async function() {
+      recorder = record(this);
+      /*Place your code here*/
+    });
+
+    afterEach(async () => {
+      /*Place your code here*/
+      recorder.stop();
+    });
+
+    it("should abort when abort() is called", async () => {
+      ...
+      ...
+    });
+  });
+  ```
+
+  (Node Test) Recording corresponding to the above test is placed at
+  `./recordings/node/aborter/recording_should_abort_when_abort_is_called.js`
+
+  [ Following this rule - `./recordings/node/<describe-block-title>/recording_<test-title>.js` ]
+
+- Just like the test recordings, we save the requests and responses from the `before` and `after` sections of the describe block in - `recordings/{node|browsers}/<describe-block-title>/recording_before_all_hook.{js|json}`.
+- In node recordings(Nock), the query parameters are not being stored and the SAS Token query parameters are not being stored in browser recordings(Nise).
+
+### ENV Variables
+
+- Any potential plain secrets in the recordings are replaced with dummy values.
+- `setReplaceableVariables` and `setReplacements` methods are being exposed from the recorder package so that the environment variables can be managed from the tests in the sdk. These two methods can be imported from `@azure/test-utils-recorder`.
+- Taking tests in the Keyvault sdk as an example -
+
+  ```typescript
+  setReplaceableVariables({
+    AZURE_CLIENT_ID: "azure_client_id",
+    AZURE_CLIENT_SECRET: "azure_client_secret",
+    AZURE_TENANT_ID: "azure_tenant_id",
+    KEYVAULT_NAME: "keyvault_name"
+  });
+  ```
+
+  Calling the `setReplaceableVariables` method would mean that
+
+  - In playback mode, the environment variables will be overriden by the provided values.
+  - In record mode, occurences of the environment variables in the recordings are replaced with the provided values.
+  - This has no effect in the `live` test mode.
+
+  ```typescript
+  setReplacements([
+    (recording: any): any =>
+      recording.replace(/"access_token":"[^"]*"/g, `"access_token":"access_token"`),
+    (recording: any): any =>
+      keySuffix === "" ? recording : recording.replace(new RegExp(keySuffix, "g"), "")
+  ]);
+  ```
+
+  Calling the `setReplacements` method would mean that
+
+  - In `record` mode, occurences of any strings in the recordings that match the regular expressions are replaced with the provided values.
+  - This has no effect in the `live` test mode or `playback` mode.
+
+- The above methods can be called from the `before` section so that the tests can leverage the environment variables.
+
+- Currently, the environment variables for storage-packages are managed by the recorder. `ACCOUNT_NAME` is replaced with `"fakestorageaccount"`, secret part of `ACCOUNT_SAS` and `ACCOUNT_KEY` are replaced with `"aaaaa"`.
```

Comment:
Storage-specific stuff should live outside of the recorder package.

Reply:
Removed